internal/handler/handler_test.go

```diff
@@ -486,3 +488,146 @@ func TestTransactionManagementService_GetTransactions(t *testing.T) {
 		})
 	}
 }
+func TestTransactionManagementService_DownloadTransaction(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
+		name  string
+		model model.GetTransaction
+		setup func() (*transactionManagementService, *http.Request)
+		want  func(recorder httptest.ResponseRecorder)
+	}{
+		{
+			name: "Success",
+			setup: func() (*transactionManagementService, *http.Request) {
+				mockLogic := mock.NewMockTransactionManagementServiceLogicIer(mockCtrl)
+				mockLogic.EXPECT().DownloadTransaction(gomock.Any(), "456").Times(1).Return(&respModel.Response{
+					Status:  http.StatusOK,
+					Message: codes.GetErr(codes.Success),
+					Data:    []byte("PDF"),
+				})
+				svc := &transactionManagementService{
+					logic: mockLogic,
+				}
+
+				r := httptest.NewRequest("GET", "/transactions/download/:123", nil)
+
+				ctx := session.SetSession(r.Context(), model.SessionStruct{UserId: "1234", Cookie: "456"})
+				return svc, r.WithContext(ctx)
+			},
+			want: func(rec httptest.ResponseRecorder) {
+				b, err := ioutil.ReadAll(rec.Body)
+				if err != nil {
+					return
+				}
+				if strings.Compare("PDF", string(b)) != 0 {
+					t.Errorf("Want: %v, Got: %v", "PDF", string(b))
+				}
+			},
+		},
+		{
+			name: "Failure:: DownloadTransaction :: err assert userid",
+			setup: func() (*transactionManagementService, *http.Request) {
+				mockLogic := mock.NewMockTransactionManagementServiceLogicIer(mockCtrl)
+				svc := &transactionManagementService{
+					logic: mockLogic,
+				}
+				r := httptest.NewRequest("GET", "/transactions/download/:123", nil)
+				ctx := session.SetSession(r.Context(), "")
+				return svc, r.WithContext(ctx)
+			},
+			want: func(rec httptest.ResponseRecorder) {
+				b, err := ioutil.ReadAll(rec.Body)
+				if err != nil {
+					return
+				}
+				var response respModel.Response
+				err = json.Unmarshal(b, &response)
+				tempResp := &respModel.Response{
+					Status:  http.StatusBadRequest,
+					Message: codes.GetErr(codes.ErrAssertUserid),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(&response, tempResp) {
+					t.Errorf("Want: %v, Got: %v", tempResp, &response)
+				}
+			},
+		},
+		{
+			name: "Failure:: DownloadTransaction :: not ok status code",
+			setup: func() (*transactionManagementService, *http.Request) {
+				mockLogic := mock.NewMockTransactionManagementServiceLogicIer(mockCtrl)
+				mockLogic.EXPECT().DownloadTransaction(gomock.Any(), gomock.Any()).Return(&respModel.Response{
```

Comment:
Validate mock with actual mocked data, sent in the request instead of `gomock.Any()`. This must be reflected for all cases.

Reply:
done